helion/_compiler/transforms/rng_transform.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+import ast
+from typing import TYPE_CHECKING
+
+import torch
+import torch.fx
+
+from ..ast_extension import create
+from ..ast_extension import create_arg
+from ..ast_extension import expr_from_string
+from ..generate_ast import statement_from_string
+from .base import TransformPass
+
+if TYPE_CHECKING:
+    from ..device_function import DeviceFunction
+    from ..host_function import HostFunction
+    from ..inductor_lowering import GraphInterpreter
+
+
+class RNGTransformPass(TransformPass):
+    def __init__(self, device_function: DeviceFunction) -> None:
+        self.rng_seed_buffer_param_name = device_function.new_var("rng_seed_buffer")
+        self.seed_index_map: dict[torch.fx.Node, int] = {}
+
+    @staticmethod
+    def has_rng_ops(host_function: HostFunction) -> bool:
+        """Check if any graph in the device IR contains RNG operations."""
+        rng_ops = {
+            torch.ops.aten.rand.default,
+            torch.ops.aten.randn.default,
+        }
+
+        for graph_info in host_function.device_ir.graphs:
+            for node in graph_info.graph.nodes:
+                if node.op == "call_function" and node.target in rng_ops:
+                    return True
+
+        return False
+
+    def get_or_allocate_seed_index(self, node: torch.fx.Node) -> int:
+        """Get the seed index for a given RNG operation node, allocating if necessary."""
+        if node not in self.seed_index_map:
+            seed_index = len(self.seed_index_map)
+            self.seed_index_map[node] = seed_index
+
+        return self.seed_index_map[node]
+
+    def add_kernel_arguments(self, args: list[ast.arg]) -> None:
+        """Add RNG seed buffer argument to kernel signature."""
+        args.append(create_arg(self.rng_seed_buffer_param_name))
+
+    def add_host_arguments(self, args: list[str]) -> None:
+        """Add RNG seed buffer argument to host function call."""
+        args.append("_rng_seed_buffer_host")
+
+    def get_host_preamble_statements(self) -> list[ast.AST]:
+        """Get statements to inject into host function preamble."""
+        import_stmt = statement_from_string(
+            "from torch._inductor import inductor_prims"
+        )
+
+        # Create host-side seed buffer with the required number of seeds
+        seed_buffer_stmt = statement_from_string(
+            f"_rng_seed_buffer_host = inductor_prims.seeds({len(self.seed_index_map)}, torch.device('cuda'))"
```

Comment:
Here we use the same approach as torch.compile (allocate one seed value per RNG op).